scripts/sanity_check_test.py

```diff
@@ -0,0 +1,139 @@
+import unittest
+import sys
+
+import sanity_check
+import git
+
+class ValidateTests(unittest.TestCase):
+    """ Tests for commit validation """
+
+    def assertFailure(self, commits, expected):
+        from StringIO import StringIO
+
+        real_commits = convert_commits(commits)
+        out = StringIO()
+
+        self.assertRaises(SystemExit, sanity_check.validate_commits, real_commits, out)
+        assert out.getvalue().strip() == expected
+
+    def assertSuccess(self, commits):
+        real_commits = convert_commits(commits)
+
+        try:
+            sanity_check.validate_commits(real_commits)
+        except:
+            print "Unexpected error:", sys.exc_info()[0]
+            raise
+
+
+    # Test cases that excercise invalid scenarios.
+
+    def test_squash_not_matching_carry(self):
+        """
+        [SQUASH] commit has a type that is not matching any [CARRY].
+        """
+
+        commits = [
```

Comment:
I was talking about the logic for the test, not for the business logic -- I mean that there is extraneous data in this list that is not necessary for the correct execution of the test. I was also asking if, in order to test this feature, we need two tests -- one that tested that if there were no `[CARRY]` commits with the same subject as the `[SQUASH]` it would fail as well as testing if there were no `[CARRY]` commits at all it would fail.